bootstrap/multi-comment-transcriptions.py

```diff
@@ -0,0 +1,182 @@
+"""A script to fix multi comment transcriptions in Blossom.
+
+Due to a bug, only the first comment of multi-comment transcriptions
+was sent to Blossom.
+"""
+import logging
+from datetime import datetime
+from typing import Dict, List
+
+from psaw import PushshiftAPI
+
+from bootstrap import END_DATE, START_DATE, blossom
+from bootstrap.migrate_redis_data import (
+    CommentData,
+    comment_filter,
+    dict_from_comment,
+    extract_id_from_grafeas_url,
+)
+
+push = PushshiftAPI()
+
+
+def _get_link_id(tr: Dict) -> str:
+    """Get the Reddit link ID of the given transcription.
+
+    For https://reddit.com/r/worldnewsvideo/comments/s48db9/bronx_fire_landlord_had_history_of_neglecting_heat/hss8wz0/
+    it returns s48db9.
+    """
+    return tr["url"].split("/")[6]
+
+
+def _remove_footer(transcription: str) -> str:
+    """Remove the footer of the transcription."""
+    parts = transcription.split("---")
+    if len(parts) < 3:
+        logging.warning(f"Weird transcription format:\n{transcription}")
+        return transcription
+
+    return "---".join(parts[:-1]).strip()
+
+
+def _join_transcriptions(transcriptions: List[CommentData]) -> str:
+    """Join a multi-comment transcription into a single text."""
+    texts = [tr["body"] for tr in transcriptions]
+    first_texts = texts[:-1]
+    # Remove the footers in the middle
+    combined_texts = [_remove_footer(txt) for txt in first_texts] + [texts[-1].strip()]
+    return "\n\n---\n\n".join(combined_texts)
+
+
+def fix_multi_comment_transcription(transcription: Dict):
+    """Fix a multi comment transcription in Blossom."""
+    if not (7000 < len(transcription["text"]) < 10050):
+        # If we have an unprocessed first comment of a multi-comment transcription,
+        # it will be rather long, but shorter than ~10000 characters (with a small buffer)
+        return
+
+    # Get the transcription author
+    tr_id = transcription["id"]
+    tr_url = transcription["url"]
+    author_id = extract_id_from_grafeas_url(transcription["author"])
+    author_response = blossom.get(f"volunteer/{author_id}")
+    if not author_response.ok:
+        logging.error(
+            f"Failed to fetch author {author_id} for transcription {tr_id} {tr_url}!"
+            f"({author_response.status_code})"
+        )
+        return
+    author = author_response.json()
+
+    # Get the transcription comments from Pushshift
+    transcriptions = list(
+        push.search_comments(
+            link_id=_get_link_id(transcription),
+            author=author["username"],
+            filter=comment_filter,
+        )
+    )
+    transcriptions: List[CommentData] = [dict_from_comment(x) for x in transcriptions]
+    # Make sure the comments are actually transcriptions
+    transcriptions = [
+        x
```

Comment:
tsk tsssskkkk

Reply:
What's wrong with this? The naming?

Reply:
I'm just giving you crap since you've dinged me for single-character variables too 😉